pkg/controller/job/job_controller_test.go

```diff
@@ -2437,6 +2437,10 @@ func TestTrackJobStatusAndRemoveFinalizers(t *testing.T) {
 	}
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
+			if !tc.enableJobBackoffLimitPerIndex {
+				// TODO: this will be removed in 1.36
+				featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, feature.DefaultFeatureGate, utilversion.MustParse("1.32"))
+			}
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobBackoffLimitPerIndex, tc.enableJobBackoffLimitPerIndex)
```

Comment:
```suggestion
```

Do we still need to perform this?
IIUC, we can delegate this to the above emulation, right?

Reply:
Yes, commenting out this line makes the tests fail. 

An alternative which works is moving this line under the `if` - next to the `SetFeatureGateEmulationVersionDuringTest` call. 

However, moving this line generates extra diff which is not necessary, so maybe better not to. Also, I followed the pattern for other feature-gates, like this one: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/statefulset/stateful_set_control_test.go#L223-L227.

Reply:
That makes sense. Thank you for confirming that!